### PR TITLE
Turned on "treat warnings as errors" to keep build clean.

### PR DESCRIPTION
**Pull request by Scotty1234**
Tidied up warnings that were now errors. Note in some cases warnings/errors disabled with pragma as need to be kept (for now).

### CODE REVIEW

File: src/ClassicUO.Assets/ClassicUO.Assets.csproj [hunk 1]
```diff
@@ -4,6 +4,14 @@
     <OutputType>Library</OutputType>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
```
Comment (andreakarasho) [inline src/ClassicUO.Assets/ClassicUO.Assets.csproj]: I think this prop can be made global putting it into:
https://github.com/ClassicUO/ClassicUO/blob/main/src/Directory.Build.props#L3-L7

File: src/ClassicUO.Assets/ClassicUO.Assets.csproj [hunk 2]
```diff
@@ -4,6 +4,14 @@
...
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
+    <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
```
Comment (andreakarasho) [inline src/ClassicUO.Assets/ClassicUO.Assets.csproj]: same

File: src/ClassicUO.Bootstrap/src/ClassicUO.Bootstrap.csproj [hunk 3]
```diff
@@ -11,6 +11,14 @@
     <PlatformTarget>x64</PlatformTarget>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
```
Comment (andreakarasho) [inline src/ClassicUO.Bootstrap/src/ClassicUO.Bootstrap.csproj]: same

File: src/ClassicUO.Bootstrap/src/ClassicUO.Bootstrap.csproj [hunk 4]
```diff
@@ -11,6 +11,14 @@
...
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
+    <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
```
Comment (andreakarasho) [inline src/ClassicUO.Bootstrap/src/ClassicUO.Bootstrap.csproj]: same

File: src/ClassicUO.Client/ClassicUO.Client.csproj [hunk 5]
```diff
@@ -90,6 +90,14 @@
     <_IsLinux Condition="'$([System.Runtime.InteropServices.RuntimeInformation]::IsOSPlatform($([System.Runtime.InteropServices.OSPlatform]::Linux)))' == 'true'">true</_IsLinux>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
```
Comment (andreakarasho) [inline src/ClassicUO.Client/ClassicUO.Client.csproj]: same

File: src/ClassicUO.Client/ClassicUO.Client.csproj [hunk 6]
```diff
@@ -90,6 +90,14 @@
...
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
+    <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
```
Comment (andreakarasho) [inline src/ClassicUO.Client/ClassicUO.Client.csproj]: same

File: src/ClassicUO.IO/ClassicUO.IO.csproj [hunk 7]
```diff
@@ -4,6 +4,14 @@
     <OutputType>Library</OutputType>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
```
Comment (andreakarasho) [inline src/ClassicUO.IO/ClassicUO.IO.csproj]: same

File: src/ClassicUO.IO/ClassicUO.IO.csproj [hunk 8]
```diff
@@ -4,6 +4,14 @@
...
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
+    <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
```
Comment (andreakarasho) [inline src/ClassicUO.IO/ClassicUO.IO.csproj]: same

File: src/ClassicUO.Renderer/ClassicUO.Renderer.csproj [hunk 9]
```diff
@@ -4,6 +4,14 @@
     <OutputType>Library</OutputType>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
```
Comment (andreakarasho) [inline src/ClassicUO.Renderer/ClassicUO.Renderer.csproj]: same

File: src/ClassicUO.Renderer/ClassicUO.Renderer.csproj [hunk 10]
```diff
@@ -4,6 +4,14 @@
...
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
+    <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
```
Comment (andreakarasho) [inline src/ClassicUO.Renderer/ClassicUO.Renderer.csproj]: same

File: src/ClassicUO.Utility/ClassicUO.Utility.csproj [hunk 11]
```diff
@@ -4,6 +4,14 @@
     <OutputType>Library</OutputType>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
```
Comment (andreakarasho) [inline src/ClassicUO.Utility/ClassicUO.Utility.csproj]: same

File: src/ClassicUO.Utility/ClassicUO.Utility.csproj [hunk 12]
```diff
@@ -4,6 +4,14 @@
...
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
+    <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
```
Comment (andreakarasho) [inline src/ClassicUO.Utility/ClassicUO.Utility.csproj]: same

File: src/ClassicUO.Assets/ClassicUO.Assets.csproj (hunk 1 above)
Comment (Scotty1234) [reply src/ClassicUO.Assets/ClassicUO.Assets.csproj]: All done, same as below.

**Pull request merged**